Submitted By: Alexander E. Patrakov
Date: 2007-09-07
Initial Package Version: 1.20.1
Origin: Upstream
Upstream status: will be part of 1.20.2
Description: Restores support for "default client" in the API.
Should fix the "Inappropriate ioctl for device" error when running brltty.

--- gpm-1.20.1/src/lib/liblow.c	2002-12-25 03:57:16.000000000 +0500
+++ gpm-1.20.1/src/lib/liblow.c	2007-09-07 15:31:43.000000000 +0600
@@ -249,7 +249,7 @@
             gpm_report(GPM_PR_OOPS,GPM_MESS_NO_MEM);
          memcpy(tty,option.consolename,strlen(option.consolename)-1);
          sprintf(&tty[strlen(option.consolename)-1],"%i",flag);
-      } else { /* use your current vc */ 
+      } else if (flag==0) { /* use your current vc */ 
          if (isatty(0)) tty = ttyname(0);             /* stdin */
          if (!tty && isatty(1)) tty = ttyname(1);     /* stdout */
          if (!tty && isatty(2)) tty = ttyname(2);     /* stderr */
@@ -265,7 +265,8 @@
          }
           
          conn->vc=atoi(&tty[strlen(option.consolename)-1]);
-      }
+      } else /* a default handler -- use console */
+        tty = strdup(option.consolename);
 
       if (gpm_consolefd == -1)
          if ((gpm_consolefd=open(tty,O_WRONLY)) < 0) {
